### ISSUE TITLE
FR: Support multiple key fields

### ISSUE THREAD

**Issue by lorensr** [Lepozepo/meteor-publish-with-relations#4]
I currently can't do a mapping with two keys, eg

```coffeescript
Transfer
  from: uid1
  to: uid2

mapping: [
  foreign_key: 'from'
  foreign_key: 'to'
  collection: Meteor.users
]
``` 

you have to duplicate the uids into an array, eg:

```
Transfer
  from: uid1
  to: uid2
  users: [uid1, uid2]
```

Perhaps this is because having multiple fields would not be performant compared to a single array field, in which case I withdraw the FR :)

**Comment by Lepozepo** [comment]
Well you definitely can't have 2 `foreign_key` lol. I think you might have misunderstood the way you use it. The correct way is `foreign_key:"users"` and you're done.

**Comment by lorensr** [comment]
Right, that's what you do when the schema has `users`, but it would be nice if the schema didn't need the `users` array, and the package was modified to work w/ 2 `foreign_key`s or 2 `key`s.

**Comment by Lepozepo** [comment]
Ah, I get it now, lol. That might not be too difficult to patch in but performance could be affected since you would end up querying per key vs a single array query.

**Comment by lorensr** [comment]
Okay, so it would just be in effect doing this:

```coffeescript
    mappings: [
      foreign_key: 'from'
      collection: Meteor.users
    , 
      foreign_key: 'to'
      collection: Meteor.users
   ]
```
(which works currently)

And doing that is not performant as an array.

**Comment by Lepozepo** [comment]
Using an array will increase performance for sure but both will perform well.
